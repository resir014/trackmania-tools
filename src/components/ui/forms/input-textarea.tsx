import * as React from 'react';

import clsx from 'clsx';

type InputTextareaProps = React.TextareaHTMLAttributes<HTMLTextAreaElement>;

export const InputTextarea = React.forwardRef<HTMLTextAreaElement, InputTextareaProps>(
  ({ className, rows = 4, ...rest }, ref) => {
    return (
      <textarea
        className={clsx(
          'shadow-sm rounded-md border-transparent bg-gray-800 hover:bg-gray-900 hover:bg-opacity-50 focus:ring-green-500 focus:border-green-500',
          className
        )}
        ref={ref}
        rows={rows}
        {...rest}
      />
    );
  }
);

InputTextarea.displayName = 'InputTextarea';
