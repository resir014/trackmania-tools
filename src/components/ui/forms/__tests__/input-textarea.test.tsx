import * as React from 'react';
import { render } from '@testing-library/react';
import { InputTextarea } from '../input-textarea';

describe('InputText', () => {
  it('renders correctly', () => {
    const { container } = render(<InputTextarea />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
