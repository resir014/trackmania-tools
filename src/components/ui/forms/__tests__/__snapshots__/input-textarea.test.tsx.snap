// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders correctly 1`] = `
<textarea
  class="shadow-sm rounded-md border-transparent bg-gray-800 hover:bg-gray-900 hover:bg-opacity-50 focus:ring-green-500 focus:border-green-500"
  rows="4"
/>
`;
